snuba/datasets/events.py

```diff
@@ -259,16 +246,21 @@ def __init__(self):
         self.__promoted_context_columns = promoted_context_columns
         self.__required_columns = required_columns
 
+        self.__tags_processor = TagColumnProcessor(
+            columns=all_columns,
+            promoted_columns=self._get_promoted_columns(),
+            column_tag_map=self._get_column_tag_map(),
+        )
+
     def default_conditions(self):
         return [
             ('deleted', '=', 0),
         ]
 
     def column_expr(self, column_name, body):
-        if NESTED_COL_EXPR_RE.match(column_name):
-            return self._tag_expr(column_name)
-        elif column_name in ['tags_key', 'tags_value']:
-            return self._tags_expr(column_name, body)
+        ret = self.__tags_processor.process_tags_expression(column_name, body)
```

Comment:
What is `ret`?

Reply:
The name `process_tags_expression` is a bit confusing since it processes contexts as well as tags.

Reply:
fair, will rename